src/tokens/erc721.ts

```diff
@@ -0,0 +1,66 @@
+import fetchQuestActionParams from '../quests/fetchQuestData.js'
+import axios from 'axios'
+import { type Address, type PublicClient } from 'viem'
+
+/**
+ * Fetches ERC721 token metadata from a given contract.
+ * @async
+ * @param {any} client The Viem client instance.
+ * @param {string} contractAddress The ERC721 contract address.
+ * @param {number} tokenId The token ID.
+ * @returns {Promise<any>} The metadata of the ERC721 token.
+ */
+export async function fetchERC721Metadata(
+  client: PublicClient,
+  contractAddress: string,
+  tokenId: number,
+): Promise<any> {
+  const tokenURI: string = await (client.readContract({
+    address: contractAddress as Address,
+    abi: [
+      'function tokenURI(uint256 tokenId) external view returns (string memory)',
+    ],
+    functionName: 'tokenURI',
+    args: [tokenId],
+  }) as Promise<string>)
+
+  const response = await axios.get(tokenURI)
+  return response.data
+}
+
+/**
+ * Fetches the media URL from ERC721 token metadata and performs basic validation or sanitization.
+ * @async
+ * @param {any} client - The Viem client instance.
+ * @param {string} contractAddress - The ERC721 contract address.
+ * @param {number} tokenId - The token ID.
+ * @returns {Promise<string | undefined>} - The media URL if available and valid.
+ */
+export async function fetchERC721Media(
+  client: PublicClient,
+  contractAddress: string,
+  tokenId: number,
+): Promise<string | undefined> {
+  const metadata = await fetchERC721Metadata(client, contractAddress, tokenId)
+  // Here, you'd add logic to sanitize or validate the URL?
```

Comment:
Ahhh did we want to actually add validation / sanitization?

Reply:
No I don't think so, was considering it but as mentioned above I think we handle this in the FE